### PR TITLE
FEATURE: regex matching for `rake posts:remap` and `rake posts:delete_word`

### PR DESCRIPTION
**Pull request by xrav3nz**
> CONTEXT & DISCUSSION: https://meta.discourse.org/t/replace-a-string-in-all-posts/48729/45?u=xrav3nz

This PR adds regex pattern matching support for `rake posts:remap` and `rake posts:delete_word`.

```ruby
> rake posts:remap['pattern', 'replace', type]
> rake posts:delete_word['pattern', type]
```

where `type` defaults to `string` if omitted.

Cheers 💪 

----

EDIT: The failed test seems to be flaky and is not relevant to this change.

### CODE REVIEW

File: lib/tasks/posts.rake [hunk 1]
```diff
@@ -42,31 +42,26 @@ end
...
-  else
+  elsif type != 'string' && type != 'regex'
     puts "ERROR: Expecting rake posts:rebake_match[pattern,type] where type is string or regex"
     exit 1
   end
 
   rebaked = 0
   total = search.count
 
-  search.find_each do |post|
```
Comment (SamSaffron) [inline lib/tasks/posts.rake]: Hmmm we stopped using find_each here so every single post gets loaded into memory, this is not really going to work on large databases.

Comment (xrav3nz) [reply lib/tasks/posts.rake]: Sorry I'm a bit lost here; this _is_ using `find_each` so not all records are loaded at once. Are you referring to the other tasks that use `each`?

Comment (SamSaffron) [reply lib/tasks/posts.rake]: see: https://apidock.com/rails/ActiveRecord/Batches/ClassMethods/find_each find_each performs the work in batchs so memory usage is low.

Comment (SamSaffron) [reply lib/tasks/posts.rake]: ahh did not notice you did a find each after the raw_match, my bad

**Comment by SamSaffron** [comment]
It looks safe, it is tested, I say we merge.  thanks!

**Pull request merged**